File: VexaDrive.Api/VexaDrive.Api/DTO/ServiceRequest/ServiceRequestDetailsDTO.cs
using System;
using System.Collections.Generic;
using VexaDriveAPI.DTO.Bill;
using VexaDriveAPI.DTO.Notification;


namespace VexaDriveAPI.DTO.ServiceRequest
{
    public class ServiceRequestDetailsDTO
    {
        public int ServiceRequestId { get; set; }
        public string CustomerUserId { get; set; }
        public int VehicleId { get; set; }
        public string VehicleModel { get; set; }
        public string ProblemDescription { get; set; }
        public string ServiceAddress { get; set; }
        public DateTime ServiceDate { get; set; }
        public string Status { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public DateTime? EstimatedDeliveryDate { get; set; }

        public BillDetailsDTO Bill { get; set; }
        public IEnumerable<NotificationListDTO> Notifications { get; set; }
    }
}

File: frontend/src/app/models/service-request/service-request-details.dto.ts
import { ServiceStatus } from './service-status.enum';
import { BillDetailsDto } from '../bill/bill-details.dto';
import { NotificationListDto } from '../notification/notification-list.dto';

export interface ServiceRequestDetailsDTO {
  serviceRequestId: number;
  customerUserId: string;
  vehicleId: number;
  vehicleModel: string;
  problemDescription: string;
  serviceAddress: string;
  serviceDate: string;
  serviceDateOnly?: string;
  status: ServiceStatus;
  createdAt: string;
  createdAtDate?: string;
  updatedAt: string;
  updatedAtDate?: string;
  estimatedDeliveryDate?: string;
  estimatedDeliveryDateOnly?: string;
  ownerName?: string;
  ownerEmail?: string;
  ownerContact?: string;
  bill?: BillDetailsDto;
  notifications: NotificationListDto[];
}
